plugins/social_network/social_network.cpp

```diff
@@ -300,44 +372,62 @@ namespace golos { namespace plugins { namespace social_network {
         o.op.visit(ovisit);
     } FC_CAPTURE_AND_RETHROW() }
 
+
     void social_network::impl::on_block(const signed_block& b) { try {
         const auto& dp = depth_parameters;
 
-        if (!dp.need_clear()) {
-            return;
-        }
+        if (dp.need_clear()) {
+            const auto& content_idx = db.get_index<comment_content_index>().indices().get<by_block_number>();
 
-        const auto& idx = db.get_index<comment_content_index>().indices().get<by_block_number>();
+            for (auto itr = content_idx.begin(); itr != content_idx.end();) {
+                auto& content = *itr;
+                ++itr;
 
-        for (auto itr = idx.begin(); itr != idx.end();) {
-            auto& content = *itr;
-            ++itr;
+                auto& comment = db.get_comment(content.comment);
 
-            auto& comment = db.get(content.comment);
+                auto delta = db.head_block_num() - content.block_number;
+                if (comment.mode == archived && dp.should_delete_part_of_content_object(delta)) {
+                    if (dp.should_delete_whole_content_object(delta)) {
+                        db.remove(content);
+                        continue;
+                    }
 
-            auto delta = db.head_block_num() - content.block_number;
-            if (comment.mode == archived && dp.should_delete_part_of_content_object(delta)) {
-                if (dp.should_delete_whole_content_object(delta)) {
-                    db.remove(content);
-                    continue;
-                }
+                    db.modify(content, [&](comment_content_object& con) {
+                        if (dp.has_comment_title_depth && delta > dp.comment_title_depth) {
+                            con.title.clear();
+                        }
 
-                db.modify(content, [&](comment_content_object& con) {
-                    if (dp.has_comment_title_depth && delta > dp.comment_title_depth) {
-                        con.title.clear();
-                    }
+                        if (dp.has_comment_body_depth && delta > dp.comment_body_depth) {
+                            con.body.clear();
+                        }
 
-                    if (dp.has_comment_body_depth && delta > dp.comment_body_depth) {
-                        con.body.clear();
-                    }
+                        if (dp.has_comment_json_metadata_depth && delta > dp.comment_json_metadata_depth) {
+                            con.json_metadata.clear();
+                        }
+                    });
 
-                    if (dp.has_comment_json_metadata_depth && delta > dp.comment_json_metadata_depth) {
-                        con.json_metadata.clear();
-                    }
-                });
+                } else {
+                    break;
+                }
+            }
+        }
 
-            } else {
-                break;
+        if (db.has_index<comment_last_update_index>()) {
```

Comment:
Function like `dp.need_clear()` should be created for case of `comment_last_update`.

Reply:
@afalaleev Why is it need? `need_clear()` is actually just "do_not_write_because_of_config_flag()". Index just not creating in such cases. has_index is enough.

Reply:
It is needed for case `!has_comment_last_update_depth`